addons/web/static/src/core/context.js

```diff
@@ -27,3 +28,25 @@ export function makeContext(contexts, initialEvaluationContext) {
     }
     return context;
 }
+
+/**
+ * Allow to evaluate a context with an incomplete evaluation context. The evaluated context only
+ * contains keys whose values are static or can be evaluated with the given evaluation context.
+ *
+ * @param {string} context
+ * @param {Object} [evaluationContext={}]
+ * @returns {Context}
+ */
+export function evalPartialContext(_context, evaluationContext = {}) {
+    const ast = parseExpr(_context);
+    const context = {};
+    for (const [key, val] of Object.entries(ast.value)) {
```

Comment:
just curious why you have to format it back to a string before evaluating it? isnt it simpler to do just evaluate the ast?
```js
    for (const key in ast.value) {
        const value = ast.value[key].val;
        try {
            context[key] = evaluate(value, evaluationContext);
        } catch {
            // ignore this key as we can't evaluate its value
        }
    }
```